temba/channels/views.py

```
@@ -1113,46 +1112,28 @@ def post(self, request, *args, **kwargs):
             channel = self.get_object()
 
             try:
-                channel.release(trigger_sync=self.request.META["SERVER_NAME"] != "testserver")
-
-                if channel.channel_type == "T" and not channel.is_delegate_sender():
-                    messages.info(
-                        request,
-                        _(
-                            "We have disconnected your Twilio number. If you do not need this number you can delete it from the Twilio website."
-                        ),
-                    )
-                else:
-                    messages.info(request, _("Your channel has been removed."))
-
-                response = HttpResponse()
-                response["Temba-Success"] = self.get_success_url()
-                return response
-
+                channel.release(request.user)
             except TwilioRestException as e:
                 messages.error(
                     request,
                     _(
-                        "Twilio reported an error removing your channel (Twilio error %s). Please try again later."
-                        % e.code
+                        f"Twilio reported an error removing your channel (error code {e.code}). Please try again later."
                     ),
                 )
 
                 response = HttpResponse()
-                response["Temba-Success"] = self.get_success_url()
+                response["Temba-Success"] = self.cancel_url
                 return response
 
-            except ValueError as e:
-                logger.error("Error removing a channel", exc_info=True)
-
-                messages.error(request, str(e))
-                return HttpResponseRedirect(reverse("channels.channel_read", args=[channel.uuid]))
-
-            except Exception:  # pragma: no cover
```

the only exception type we don't swallow is `TwilioRestException` so these except blocks aren't actually used